### PR TITLE
Fix issue #89: Read recipient from encoded hexadecimal notation

### PR DESCRIPTION
**Pull request by evias**
### Added

- Added class `Recipient` with `createFromEncoded()`
- Added `NamespaceId.createFromEncoded()`
- Added `TransactionTransaction.recipient` capability to hold `Recipient | Address | NamespaceId`
- Added unit tests for encoded hexadecimal notation interpretation

### Fixed

- Fixed `infrastructure/transaction/CreateTransactionFromDTO` for TransferTransaction to parse recipient correctly. (#89 )

### CODE REVIEW

**Comment by evias** [comment]
Removed proposed class `Recipient` in favor of an auxiliary function `extractVersion()` in `transaction/CreateTransactionFromDTO`.

**Comment by evias** [comment]
Added auxiliary `extractMosaics()` to differentiate mosaicId from namespaceId in `mosaics`

File: src/infrastructure/transaction/CreateTransactionFromDTO.ts [hunk 1]
```diff
@@ -339,3 +339,42 @@ const extractNetworkType = (version: number): NetworkType => {
...
+            // only 8 bytes are relevant to resolve the NamespaceId
+            const relevantPart = recipient.substr(2, 16);
+            return NamespaceId.createFromEncoded(relevantPart);
+        }
+
+        // read address from encoded hexadecimal notation
+        return Address.createFromEncoded(recipient);
+};
+
+const extractMosaics = (mosaics: any): Mosaic[] => {
```
Comment (rg911) [inline src/infrastructure/transaction/CreateTransactionFromDTO.ts]: Perhaps provide a short description in the document for this private method? as there are some interesting logic here, e.g. the `most significant bit of byte 0 is set`

File: src/infrastructure/transaction/CreateTransactionFromDTO.ts [hunk 2]
```diff
@@ -339,3 +339,42 @@ const extractNetworkType = (version: number): NetworkType => {
 const extractTransactionVersion = (version: number): number => {
     return parseInt(version.toString(16).substr(2, 2), 16);
 };
+
+const extractRecipient = (recipient: string): Address | NamespaceId => {
```
Comment (rg911) [inline src/infrastructure/transaction/CreateTransactionFromDTO.ts]: Perhaps provide a short description in the document for this private method?

**Pull request merged**